ixmp/core.py

```diff
@@ -520,6 +535,15 @@ def add_timeseries(self, df, meta=False):
         # Columns (year) as integer
         df.columns = df.columns.astype(int)
 
+        # Identify columns to drop
+        to_drop = set()
+        if year_lim[0]:
+            to_drop |= set(filter(lambda y: y < year_lim[0], df.columns))
+        if year_lim[1]:
+            to_drop |= set(filter(lambda y: y > year_lim[1], df.columns))
+
+        df.drop(to_drop, axis=1, inplace=True)
```

Comment:
nit: maybe avoid operation if to_drop is empty?

Reply:
I think it's a no-op and not very expensive.